woocommerce/woocommerce
# Added upload plugin functionality

### Changes proposed in this Pull Request:

The goal of this PR is to allow installing the WooCommerce plugin. It makes the following changes:

* updates to `smoke-test-daily.yml` to run the plugin upload utility for WooCommerce
* `get-plugin-zip.js` in the environment package for handling getting the zip and checking if it has a nested zip.
* three utility methods: `getSlug()` to get the slug string for a plugin; `describeIf()` to conditionally run a test suite; `itIf()` to conditionally run a test case.
* merchant workflows around plugin work: `uploadAndActivatePlugin()`, `activatePlugin()`, `deactivatePlugin()`, `deletePlugin()`

### How to test the changes in this Pull Request:

1. Review changes and the updates to the README and change log files
2. Run `npm install`
2. Run `npx wc-e2e docker:up`
3. Run `npx wc-e2e test:e2e`
4. Verify tests pass

##### Test plugin workflow

* Update `update-woocommerce.test.js` with new values for:
 * `nightlyZip`
 * `pluginName`
 * For example, in my testing I used:
 
```
const nightlyZip = 'https://github.com/Automattic/woocommerce-payments/releases/download/2.6.1/woocommerce-payments.zip';
const pluginName = 'WooCommerce Payments';
```

⚠️ Make sure the above is done first before running the command below, otherwise this will break the local test environment.

After the above is done, run the test either in headless or non-headless mode:

* `npx wc-e2e test:e2e ./tests/e2e/specs/smoke-tests/update-woocommerce.test.js`
* `npx wc-e2e test:e2e-dev ./tests/e2e/specs/smoke-tests/update-woocommerce.test.js`

Then:

* Verify the tests pass
* Login to the test environment and navigate to `Plugins > Installed Plugins` page
* Verify the plugin you defined in the setup above has been installed and activated as expected
